src/stan/error_handling/matrix/check_multiplicable.hpp

```diff
@@ -3,23 +3,46 @@
 
 #include <sstream>
 #include <stan/meta/traits.hpp>
-#include <stan/error_handling/scalar/dom_err.hpp>
 #include <stan/math/matrix/Eigen.hpp>
 #include <stan/error_handling/matrix/check_size_match.hpp>
+#include <stan/error_handling/scalar/check_positive_index.hpp>
 
 namespace stan {
   namespace error_handling {
 
-    // NOTE: this will not throw if y1 or y2 contains nan values.
+    /**
+     * Return <code>true</code> if the matrices can be multiplied.
+     *
+     * This checks the runtime sizes to determine whether the two
+     * matrices are multiplicable. This allows Eigen matrices,
+     * vectors, and row vectors to be checked.  This function does not
+     * check the values within the matrices for <code>NaN</code>.
```

Comment:
I don't think we need to include information about what functions don't do, such as not checking for NaN.